tools/run_clippy.sh

```diff
@@ -37,4 +34,123 @@ CLIPPY_ARGS="
 -D clippy::wildcard_imports
 "
 
-cargo clippy -- $CLIPPY_ARGS
+# Disallow all complexity lints, then re-allow each one Tock does not comply
+# with.
+#
+# There are three sections:
+# 1. The first section are lints we almost certainly don't want.
+# 2. The section section are lints we may not want, we probably have to see the
+#    resulting diff.
+# 3. The third section are lints that we do want we just need to fixup the code
+#    to pass the lint checks.
+CLIPPY_ARGS_COMPLEXITY="
+-D clippy::complexity
```

Comment:
I might put the complexity link on a separate line, at a glance it looks like you accidentally denied one of the lints from the first section that you meant to allow